### ISSUE TITLE
[minecraft] Setting up GID and umask

### ISSUE THREAD

**Issue by rrauenza** [itzg/dockerfiles#55]
I've been making some modifications to the scripts because I'd like to make sure the container is deployed with a specific GID -- that way I can put the minecraft admins (outside the container) inside of a group that has write access to "/data"

So I added this to start.sh,

```
 set -e
 usermod --uid $UID minecraft
-chown -R minecraft /data /start-minecraft
+
+if [ -n "$GID" ]
+then
+    # Patch the GID to what we specified.
+    groupmod --gid $GID minecraft
+fi
+
+# Make group writable if we've specified a specific GID
+if [ -n "$GID" ]
+then
+    chown -R minecraft:$GID /data /start-minecraft
+    chmod -R g+wX /data /start-minecraft
+    umask 002
+else
+    chown -R minecraft /data /start-minecraft
+    umask 022
+fi
```

But the umask also needs to be done in start-minecraft.sh,

```
 #!/bin/bash
 
+if [ -n "$GID" ]
+then
+    umask 002
+else
+    umask 022
+fi
+
 if [ ! -e /data/eula.txt ]; then
```

I was going to do a pull request, but I wonder if I should actually just do another environment variable for UMASK rather than checking whether GID is set.  I'm ok with the way it is, but if you want my code, I can write it the way you'd like to see it.

This also may require some reordering and changes in Dockerfile...

**Comment by itzg** [comment]
I think that's a great idea, so definitely move forward with a PR. As for the umask, I think deriving it from the presence of GID makes sense. 

As an aside I still wish Docker had some better support for mapping host to container users. Also, you might find that doing a 'docker exec' into the container to edit files and 'docker cp' to bring files in and out might suffice...but not really kid friendly. 

I have an adjacent project https://github.com/itzg/minecraft-container-yard where maintaining server files is a feature I hope to add.

**Comment by rrauenza** [comment]
Great!  Glad you like the idea -- I'm probably going to fork off my own now, though, partly to learn more about Docker, and partly because I'd rather have a CentOS container.  I'll continue to feedback ideas back into yours if you like.

One change I did is move away from jsawk to jq, since it is an installable package:

```
$ git diff | grep jq
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y libmozjs-24-bin imagemagick lsof curl jq && apt-get clean
+    VANILLA_VERSION=`curl --silent https://s3.amazonaws.com/Minecraft.Download/versions/versions.json | jq -r '.latest.release'`
+    VANILLA_VERSION=`curl --silent https://s3.amazonaws.com/Minecraft.Download/versions/versions.json | jq -r '.latest.snapshot'`
+    VANILLA_VERSION=`curl --silent https://s3.amazonaws.com/Minecraft.Download/versions/versions.json | jq -r '.latest.release'`
+            FORGE_VERSION=`curl --silent http://files.minecraftforge.net/maven/net/minecraftforge/forge/promotions_slim.json | jq -r  ".promos.\"$norm-recommended\""`
```

I also switched to curl because by default wget makes backups.

I specifically didn't want to give him docker access directly since docker privileges can fairly easily be used to gain root outside the container (create a container, mount a volume, drop a setuid file, ...)

http://reventlov.com/advisories/using-the-docker-command-to-root-the-host

**Comment by itzg** [comment]
Actually, I'd love to switch to `jq` and `curl`. In fact, somewhere in my local repo I actually made the conversion to `jq`. I have much more experience with that tool vs the 5 minutes experience I gained on `jsawk` in the Minecraft script. So I'd be happy for contributions of all of the above.

Now a `FROM` switch to CentOS gets complicated :).

**Comment by itzg** [comment]
Resolved by #57
